text/3729-sized-hierarchy.md

```diff
@@ -0,0 +1,1412 @@
+- Feature Name: `sized_hierarchy`
+- Start Date: 2024-09-30
+- RFC PR: [rust-lang/rfcs#3729](https://github.com/rust-lang/rfcs/pull/3729)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+All of Rust's types are either *sized*, which implement the `Sized` trait and
+have a statically known size during compilation, or *unsized*, which do not
+implement the `Sized` trait and are assumed to have a size which can be computed
+at runtime. However, this dichotomy misses two categories of type - types whose
+size is unknown during compilation but is a runtime constant, and types
+whose size can never be known. Supporting the former is a prerequisite to stable
+scalable vector types and supporting the latter is a prerequisite to unblocking
+extern types.
+
+This RFC necessarily depends on the experimental feature
+[`const Trait`][goal_const_traits] and is written assuming familiarity with that
+feature.
```

Comment:
I thought I knew the feature quite well, but it is not clear to me why traits need to be specially marked to allow "const" trait bounds. Is that explained anywhere? Sadly the link placed here doesn't go anywhere that explains anything.

Reply:
I believe it's just that if a trait isn't marked as const then it doesn't permit const implementations and as such it doesn't make sense to have const bounds on it, but I'm likewise making educated guesses from what I've read and understood from speaking to those working on const traits.